test/integration/serviceaccount/service_account_test.go

```diff
@@ -261,62 +266,42 @@ func TestServiceAccountTokenAuthentication(t *testing.T) {
 	}
 
 	// Create "ro" user in myns
-	_, err = c.CoreV1().ServiceAccounts(myns).Create(context.TODO(), &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: readOnlyServiceAccountName}}, metav1.CreateOptions{})
+	roSA, err := c.CoreV1().ServiceAccounts(myns).Create(context.TODO(), &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: readOnlyServiceAccountName}}, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Service Account not created: %v", err)
 	}
-	roTokenName, roToken, err := getReferencedServiceAccountToken(c, myns, readOnlyServiceAccountName, true)
+
+	roTokenName := "ro-test-token"
+	secret, err := createServiceAccountToken(c, roSA, myns, roTokenName)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("Secret not created: %v", err)
 	}
 	roClientConfig := config
-	roClientConfig.BearerToken = roToken
+	roClientConfig.BearerToken = string(secret.Data[v1.ServiceAccountTokenKey])
 	roClient := clientset.NewForConfigOrDie(&roClientConfig)
 	doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
 	doServiceAccountAPIRequests(t, roClient, otherns, true, false, false)
 	err = c.CoreV1().Secrets(myns).Delete(context.TODO(), roTokenName, metav1.DeleteOptions{})
 	if err != nil {
 		t.Fatalf("could not delete token: %v", err)
 	}
-	// wait for delete to be observed and reacted to via watch
-	wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
```

Comment:
It's possible dropping this delay made us proceed to the next check of the deleted token too quickly. Check if the 10-second token cache is used in this test's authenticator. If so, we may need a wait loop for up to 10 seconds while the token expires from the auth cache before proceeding

Reply:
Doesn't look like the cache is used here, but the secret deletion would need to be observed by the authenticator. Should probably restore a brief wait loop here for the credential to be invalidated

Reply:
opened https://github.com/kubernetes/kubernetes/pull/108507